lib/Page.js

```diff
@@ -269,6 +270,26 @@ class Page extends EventEmitter {
     return context.evaluateHandle(pageFunction, ...args);
   }
 
+  /**
+   * @param {string} contextIdentifier
+   * @returns {!Promise.<!Puppeteer.ExecutionContext>}
+   */
+  async createNewIsolatedContext(contextIdentifier) {
+    const mainFrame = this.mainFrame();
+    const isolatedWorldInfo = await this._client.send(
+        'Page.createIsolatedWorld',
+        {frameId: mainFrame._id, worldName: contextIdentifier}
```

Comment:
This looks like it should be a frame-level method. Then the page-level method would just do `this.mainFrame().createNewIsolatedContext()`

Reply:
Happy to do it, but I'm not sure how. To create a context I need a JSHandleFactory which I can only get from a FrameManager. That I can access from Page but not from Frame. Any idea how to deal with that?

I refer to:

`
    const JsHandleFactory = this._frameManager.createJSHandle.bind(this._frameManager, executionContextId);
`

Which is a required parameter to create a new execution context.

Reply:
@furstenheim you'll need to await the `ExecutionContext` with this id; I'd do this after the  #2812 lands.